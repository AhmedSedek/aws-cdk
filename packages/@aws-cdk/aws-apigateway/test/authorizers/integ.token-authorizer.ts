import lambda = require('@aws-cdk/aws-lambda');
import { App, Stack } from '@aws-cdk/core';
import path = require('path');
import { AuthorizationType, LambdaTokenAuthorizer, MockIntegration, PassthroughBehavior, RestApi } from '../../lib';

const app = new App();
const stack = new Stack(app, 'MyStack');

const authorizerFn = new lambda.Function(stack, 'MyAuthorizerFunction', {
  runtime: lambda.Runtime.NODEJS_10_X,
  handler: 'index.handler',
  code: lambda.AssetCode.fromAsset(path.join(__dirname, 'integ.token-authorizer.handler'))
});

const restapi = new RestApi(stack, 'MyRestApi');

const authorizer = new LambdaTokenAuthorizer(stack, 'MyAuthorizer', {
  function: authorizerFn,
  restApi: restapi,
  headerName: 'x-auth-header'
});

restapi.root.addMethod('ANY', new MockIntegration({
  integrationResponses: [
    { statusCode: '200' }
  ],
  passthroughBehavior: PassthroughBehavior.NEVER,
  requestTemplates: {
    'application/json': '{ "statusCode": 200 }',
  },
}), {
  methodResponses: [
    { statusCode: '200' }
  ],
  authorizer,
  authorizationType: AuthorizationType.CUSTOM
});