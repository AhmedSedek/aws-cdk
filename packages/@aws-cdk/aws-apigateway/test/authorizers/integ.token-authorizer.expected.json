{
  "Resources": {
    "MyAuthorizerFunctionServiceRole8A34C19E": {
      "Type": "AWS::IAM::Role",
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com"
              }
            }
          ],
          "Version": "2012-10-17"
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition"
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole"
              ]
            ]
          }
        ]
      }
    },
    "MyAuthorizerFunction70F1223E": {
      "Type": "AWS::Lambda::Function",
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Ref": "AssetParametersa2c916cc1a85194847a0427ac18e4e61a8faae242b4a23f1b4d4059249197b91S3BucketA67237B6"
          },
          "S3Key": {
            "Fn::Join": [
              "",
              [
                {
                  "Fn::Select": [
                    0,
                    {
                      "Fn::Split": [
                        "||",
                        {
                          "Ref": "AssetParametersa2c916cc1a85194847a0427ac18e4e61a8faae242b4a23f1b4d4059249197b91S3VersionKey97F8B451"
                        }
                      ]
                    }
                  ]
                },
                {
                  "Fn::Select": [
                    1,
                    {
                      "Fn::Split": [
                        "||",
                        {
                          "Ref": "AssetParametersa2c916cc1a85194847a0427ac18e4e61a8faae242b4a23f1b4d4059249197b91S3VersionKey97F8B451"
                        }
                      ]
                    }
                  ]
                }
              ]
            ]
          }
        },
        "Handler": "index.handler",
        "Role": {
          "Fn::GetAtt": [
            "MyAuthorizerFunctionServiceRole8A34C19E",
            "Arn"
          ]
        },
        "Runtime": "nodejs10.x"
      },
      "DependsOn": [
        "MyAuthorizerFunctionServiceRole8A34C19E"
      ]
    },
    "MyRestApi2D1F47A9": {
      "Type": "AWS::ApiGateway::RestApi",
      "Properties": {
        "Name": "MyRestApi"
      }
    },
    "MyRestApiDeploymentB555B5828fad37a0e56bbac79ae37ae990881dca": {
      "Type": "AWS::ApiGateway::Deployment",
      "Properties": {
        "RestApiId": {
          "Ref": "MyRestApi2D1F47A9"
        },
        "Description": "Automatically created by the RestApi construct"
      },
      "DependsOn": [
        "MyRestApiANY05143F93"
      ]
    },
    "MyRestApiDeploymentStageprodC33B8E5F": {
      "Type": "AWS::ApiGateway::Stage",
      "Properties": {
        "RestApiId": {
          "Ref": "MyRestApi2D1F47A9"
        },
        "DeploymentId": {
          "Ref": "MyRestApiDeploymentB555B5828fad37a0e56bbac79ae37ae990881dca"
        },
        "StageName": "prod"
      }
    },
    "MyRestApiCloudWatchRoleD4042E8E": {
      "Type": "AWS::IAM::Role",
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "apigateway.amazonaws.com"
              }
            }
          ],
          "Version": "2012-10-17"
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition"
                },
                ":iam::aws:policy/service-role/AmazonAPIGatewayPushToCloudWatchLogs"
              ]
            ]
          }
        ]
      }
    },
    "MyRestApiAccount2FB6DB7A": {
      "Type": "AWS::ApiGateway::Account",
      "Properties": {
        "CloudWatchRoleArn": {
          "Fn::GetAtt": [
            "MyRestApiCloudWatchRoleD4042E8E",
            "Arn"
          ]
        }
      },
      "DependsOn": [
        "MyRestApi2D1F47A9"
      ]
    },
    "MyRestApiANY05143F93": {
      "Type": "AWS::ApiGateway::Method",
      "Properties": {
        "HttpMethod": "ANY",
        "ResourceId": {
          "Fn::GetAtt": [
            "MyRestApi2D1F47A9",
            "RootResourceId"
          ]
        },
        "RestApiId": {
          "Ref": "MyRestApi2D1F47A9"
        },
        "AuthorizationType": "CUSTOM",
        "AuthorizerId": {
          "Ref": "MyAuthorizer6575980E"
        },
        "Integration": {
          "IntegrationResponses": [
            {
              "StatusCode": "200"
            }
          ],
          "PassthroughBehavior": "NEVER",
          "RequestTemplates": {
            "application/json": "{ \"statusCode\": 200 }"
          },
          "Type": "MOCK"
        },
        "MethodResponses": [
          {
            "StatusCode": "200"
          }
        ]
      }
    },
    "MyAuthorizer6575980E": {
      "Type": "AWS::ApiGateway::Authorizer",
      "Properties": {
        "RestApiId": {
          "Ref": "MyRestApi2D1F47A9"
        },
        "Type": "TOKEN",
        "AuthorizerUri": {
          "Fn::Join": [
            "",
            [
              "arn:aws:apigateway:",
              {
                "Ref": "AWS::Region"
              },
              ":lambda:path/2015-03-31/functions/",
              {
                "Fn::GetAtt": [
                  "MyAuthorizerFunction70F1223E",
                  "Arn"
                ]
              },
              "/invocations"
            ]
          ]
        },
        "IdentitySource": "method.request.header.x-auth-header",
        "Name": "MyStackMyAuthorizerCAFE6371"
      }
    },
    "MyAuthorizerMyStackMyAuthorizerCAFE6371Permissions168057A2": {
      "Type": "AWS::Lambda::Permission",
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "MyAuthorizerFunction70F1223E",
            "Arn"
          ]
        },
        "Principal": "apigateway.amazonaws.com"
      }
    }
  },
  "Parameters": {
    "AssetParametersa2c916cc1a85194847a0427ac18e4e61a8faae242b4a23f1b4d4059249197b91S3BucketA67237B6": {
      "Type": "String",
      "Description": "S3 bucket for asset \"a2c916cc1a85194847a0427ac18e4e61a8faae242b4a23f1b4d4059249197b91\""
    },
    "AssetParametersa2c916cc1a85194847a0427ac18e4e61a8faae242b4a23f1b4d4059249197b91S3VersionKey97F8B451": {
      "Type": "String",
      "Description": "S3 key for asset version \"a2c916cc1a85194847a0427ac18e4e61a8faae242b4a23f1b4d4059249197b91\""
    },
    "AssetParametersa2c916cc1a85194847a0427ac18e4e61a8faae242b4a23f1b4d4059249197b91ArtifactHashB4995EF2": {
      "Type": "String",
      "Description": "Artifact hash for asset \"a2c916cc1a85194847a0427ac18e4e61a8faae242b4a23f1b4d4059249197b91\""
    }
  },
  "Outputs": {
    "MyRestApiEndpoint4C55E4CB": {
      "Value": {
        "Fn::Join": [
          "",
          [
            "https://",
            {
              "Ref": "MyRestApi2D1F47A9"
            },
            ".execute-api.",
            {
              "Ref": "AWS::Region"
            },
            ".",
            {
              "Ref": "AWS::URLSuffix"
            },
            "/",
            {
              "Ref": "MyRestApiDeploymentStageprodC33B8E5F"
            },
            "/"
          ]
        ]
      }
    }
  }
}